packages/grafana-ui/src/components/Flex/Flex.mdx

```diff
@@ -0,0 +1,123 @@
+the decision to create it is to ensure consistency in design across Grafana.
+
+### Usage
+
+#### When to use
+
+Use when in need to align components and small parts of the application. Use as parent container to wrap elements that
+you wish to align in a certain way.
+
+#### When not to use
+
+When you need to lay out bigger parts of the application or when you want to create page lay out.
+
+### Variants
+
+Flex component has few variants that can be used based on the desire alignment you need for your case.
+
+Some examples of how to use the Flex component can be seen below:
+
+- AlignItems stretch
+
+```ts
+import { Flex } from '@grafana/ui'
+
+<header>
+    <h1>Using Flex component to align-items stretch and justify-content to be center</h1>
+</header>
+<Flex direction="row" wrap="wrap"  alignItems="stretch" justifyContent="center" gap={2}>
+    <Item color="pink" height="10em" />
+    <Item color="darkmagenta" />
+    <Item color="pink" height="3em" />
+    <Item color="darkmagenta" />
+</Flex>
+```
+
+- Wrap items wrap-reverse
+
+```ts
+import { Flex } from '@grafana/ui'
+
+<header>
+    <h1>Using Flex component to align-items with wrap-reverse property</h1>
+</header>
+<Flex direction="row" wrap="wrap-reverse"  alignItems="center" justifyContent="center" gap={4}>
+    <Item color="pink"  />
+    <Item color="darkmagenta" />
+    <Item color="pink" />
+    <Item color="darkmagenta" />
+ </Flex>
+```
+
+- JustifyContent flex-start
+
+```ts
+import { Flex } from '@grafana/ui'
+
+<header>
+    <h1>Using Flex component to align-items with justify-content property</h1>
+</header>
+<Flex direction="row" wrap="wrap"  alignItems="center" justifyContent="flex-start" gap={2}>
+    <Item color="pink"  />
+    <Item color="darkmagenta" />
+    <Item color="pink" />
+    <Item color="darkmagenta" />
+ </Flex>
+```
+
+- Gap of 16px using the ThemeSpacingTokens
+
+```ts
+import { Flex } from '@grafana/ui'
+
+<header>
+    <h1>Using Flex component to align-items with gap of 16px</h1>
+</header>
+<Flex direction="row" wrap="wrap"  alignItems="center" justifyContent="center" gap={2}>
+    <Item color="pink"  />
+    <Item color="darkmagenta" />
+    <Item color="pink" />
+    <Item color="darkmagenta" />
+ </Flex>
+```
+
+- Direction column
+
+```ts
+import { Flex } from '@grafana/ui'
+
+<header>
+    <h1>Using Flex component to align-items with direction column</h1>
+</header>
+<Flex direction="column" wrap="wrap"  alignItems="center" justifyContent="center" gap={2}>
+    <Item color="pink"  />
+    <Item color="darkmagenta" />
+    <Item color="pink" />
+    <Item color="darkmagenta" />
+ </Flex>
+```
+
+### Dos
+
+    * to display the direction of the elements
+    * to set the elements to wrap
+    * to align items (center, vertically or horizontally)
+
+### Don'ts
+
+    * for complex grid lay outs with various rows and columns
+    * bidirectional lay outs
+    * complex nesting
+    * equal height columns
```

Comment:
Don't think these are needed, we can just keep when to use / when not to use and expand that. Like mention that we want to use it when we need to stack components vertically or horizontally or when we need elements to be aligned in a certain way; and not to use when we need complex nesting such as a grid layout.